src/printer.js

```diff
@@ -181,6 +181,18 @@ function genericPrintNoParens(path, options, print, args) {
     case "EmptyStatement":
       return "";
     case "ExpressionStatement":
+      // Detect Flow-parsed directives
+      if (n.directive) {
+        return concat([
+          nodeStr(
+            n,
```

Comment:
Oh, that's much better. Fixed in https://github.com/prettier/prettier/pull/1571/commits/527324f5fff2cc2185fa009d244af29e4e067ffa